ansible/library/interface_facts.py

```diff
@@ -66,19 +74,20 @@ def get_default_interfaces(ip_path, module):
     )
     interface = dict(v4 = {}, v6 = {})
     for key in command.keys():
-        """
-        if key == 'v6' and self.facts['os_family'] == 'RedHat' \
+        '''
+        if key == 'v6' and self.facts['os_family'] == 'RedHat' \\
             and self.facts['distribution_version'].startswith('4.'):
             continue
-        """
+        '''
         if key == 'v6' and not socket.has_ipv6:
             continue
-        rc, out, err = module.run_command(command[key])
+        proc = subprocess.Popen(command[key], shell=False, stdout=subprocess.PIPE)
+        (out, err) = proc.communicate()
         if not out:
             # v6 routing may result in
             #   RTNETLINK answers: Invalid argument
             continue
-        words = out.split('\n')[0].split()
+        words = out.split('\\n')[0].split()
```

Comment:
why?

Reply:
Since this code is defined as a string, the additional '\' is to escape the special meaning of '\n'